kedro/framework/cli/starters.py

```diff
@@ -93,6 +93,24 @@ class KedroStarterSpec:  # noqa: too-few-public-methods
     "An optional directory inside the repository where the starter resides."
 )
 
+# TODO; Insert actual link to the documentation.
+ADDON_ARG_HELP = """
+Select which add-ons you'd like to include. By default, none are included.
+Visit: kedro.org/{insert-documentation} to find out more about these add-ons.
+
+Add-Ons
+1) Linting : Provides a basic linting setup with Black and Ruff
+2) Testing : Provides basic testing setup with pytest
+3) Custom Logging : Provides more logging options
+4) Documentation: Basic documentation setup with Sphinx
+5) Data Structure: Provides a directory structure for storing data
+
+Example usage:
+kedro new --addons=lint,test,log,docs,data (or any subset of these options)
```

Comment:
Wouldn't we be specifying the number selection here instead?

Reply:
@amandakys designed it as implemented, so not with numbers. But it would be good to get her confirmation!

Reply:
Worried about the inconsistency between specifying by flag (through add-on name) and through the prompting (through add-on number), maybe we can look at changing the latter post-#2987?

Reply:
The numbers will be very hard to remember though, so from a usability point of view I prefer the way it's done now. But as I said, this is definitely a design detail, so @amandakys has the final say 🙂